plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/clients/FeatureDevClient.kt

```diff
@@ -36,11 +42,40 @@ import software.aws.toolkits.jetbrains.core.credentials.ToolkitConnectionManager
 import software.aws.toolkits.jetbrains.core.credentials.pinning.QConnection
 import software.aws.toolkits.jetbrains.services.amazonq.clients.AmazonQStreamingClient
 import software.aws.toolkits.jetbrains.services.codemodernizer.utils.calculateTotalLatency
+import software.aws.toolkits.jetbrains.services.telemetry.ClientMetadata
+import software.aws.toolkits.jetbrains.settings.AwsSettings
 import java.time.Instant
 import software.amazon.awssdk.services.codewhispererruntime.model.ChatTriggerType as SyncChatTriggerType
 
 @Service(Service.Level.PROJECT)
 class FeatureDevClient(private val project: Project) {
+    val FEATURE_EVALUATION_PRODUCT_NAME = "FeatureDev";
+
+    fun getTelemetryOptOutPreference() =
+        if (AwsSettings.getInstance().isTelemetryEnabled) {
+            OptOutPreference.OPTIN
+        } else {
+            OptOutPreference.OPTOUT
+        }
+
+    private val featureDevUserContext = ClientMetadata.getDefault().let {
```

Comment:
[question]: Isn't `CodeWhispererRuntimeClient` adding those fields for us? Should all `features` duplicate these part? I think in VsCode, those were added by default.

Reply:
I'll double check if it is possible to get rid of adding it to our client, but at first sight, I noticed that it wouldn't

Reply:
@sarayev In VScode those are added by default yes, but not in JetBrains. The client is auto built here

```FeatureDevClient.kt
 82 private fun bearerClient() = connection().getConnectionSettings().awsClient<CodeWhispererRuntimeClient>()
```

Codewhisperer has their own deprecated client to send telemetry events